src/components/ui/DatePicker/Day.svelte

```diff
@@ -0,0 +1,99 @@
+<script lang="ts">
+  import { getDoy } from '../../../utilities/time';
+
+  import { createEventDispatcher } from 'svelte';
+
+  export let date: Date;
+  export let maxDate: Date;
+  export let minDate: Date;
+  export let month: number;
+  export let selectedDate: Date | null = null;
+
+  let isSelected: boolean = false;
+  let isWithinBounds: boolean = true;
+
+  const dispatch = createEventDispatcher();
+
+  let isToday: boolean = false;
+
+  $: isToday = isSameDay(date, new Date());
+  $: isSelected = isSameDay(date, selectedDate);
+  $: isWithinBounds = minDate.getTime() <= date.getTime() && date.getTime() <= maxDate.getTime();
+  $: isOutsideCurrentMonth = month !== date.getUTCMonth();
+
+  function isSameDay(date1: Date, date2: Date) {
+    if (date1 && date2) {
+      return (
+        date1.getUTCDate() === date2.getUTCDate() &&
+        date1.getUTCMonth() === date2.getUTCMonth() &&
+        date1.getUTCFullYear() === date2.getUTCFullYear()
+      );
+    }
+    return false;
+  }
+
+  function onSelect() {
+    if (isWithinBounds) {
+      dispatch('select', date);
+    }
+  }
+</script>
+
+<div
+  class={`date-picker-day${isToday ? ' today' : ''}${isSelected ? ' selected' : ''}${
```

Comment:
Not required but this might be more readable if this class is built in a reactive expression

Reply:
I can write a util function that basically does what the `classnames` library does. Unless we wanted to actually use `classnames`? I'm down for either.